### ISSUE TITLE
Great idea!

### ISSUE THREAD

**Issue by mattveraldi** [Naapperas/pod#6]
I was looking exactly for this, but apparently no one did it before. Thanks.

Is there a plan for test coverage and documentation?

**Comment by Naapperas** [comment]
> I was looking exactly for this, but apparently no one did it before. Thanks.
> 
> Is there a plan for test coverage and documentation?

Hi.

I haven't been able to take time off college for this but yes, there are plans for test coverage and documentation. I wanted to change a bit how `zon` is written to better mirror `zod`'s API so somethings will change.

There are some tests already, just need to generate coverage for those. Documentation will be added in time.

If you want to contribute to those features you are more thank welcome :))

**Comment by Naapperas** [comment]
@mattveraldi while not merged with the default branch, I have added test coverage to `zon`. You can go to https://coveralls.io/github/Naapperas/zon and see the coverage information. This info will appear on the README once #8 is merged. Regarding documentation, `readthedocs` wouldn't let me create a project called `zon`, will investigate further.

**Comment by winstxnhdw** [comment]
@Naapperas How does this project differ from more popular validators like Pydantic or msgspec?

**Comment by Naapperas** [comment]
> @Naapperas How does this project differ from more popular validators like Pydantic or msgspec?

The most important difference between these projects is that `zon` was started as a hobby side-project :)

I have been asked that a lot regarding Pydantic and what I always say is that `zon` more closely follows `zod`'s API (the initial goal of the project actually). Other than that, there is less code involved so it might be more lightweight.

I did not know about `msgspec` so I took a look. `zon` has the "advantage" of being written in pure Python.

Basically `Pydantic` and `msgspec` are more powerful than `zon` so if you need that kind of control definitely use those libraries. `zon` as a library was just meant to be a lightweight port of `zod` to Python.
